File: 121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = prices[0]
        sell = prices[0]
        best = 0
        for p in prices: 
            if p < buy:
                buy = p
                sell = p
            sell = max(sell, p)
            best = max(best, sell-buy)
            
        return best 

File: 121.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int m = 0, buy = prices[0]; 
        for (int i =  1; i < prices.size(); i++) { 
            if (prices[i] < buy)  buy = prices[i]; 
            m = max(prices[i] - buy, m); 
        } 

        return m; 
    }
}
